runtime/router.go

```diff
@@ -160,7 +166,15 @@ func NewRouterEndpoint(
 func (endpoint *RouterEndpoint) HandleRequest(
 	w http.ResponseWriter, r *http.Request, params httprouter.Params,
 ) {
+	reqFields := logRequestFields(r)
+	resFields := make([]zapcore.Field, 0)
+
+	defer func() {
+		writeLogs(endpoint.gateway.Logger, reqFields, resFields)
```

Comment:
yeah, but I think it makes the code less readable. A defered anonymous function makes it clear that the arguments are modified in the execution of the containing function. It also makes the append messy to use in the build field methods.